libvast/vast/filesystem.hpp

```diff
@@ -253,7 +251,7 @@ class file {
 
 private:
   native_type handle_;
-  bool close_on_destruction_ = !close_on_destruction;
+  bool close_on_destruction_ = true;
```

Comment:
Why did you not remove the member outright?

Reply:
As far as I can tell, we can go even further and remove the `native_type` handling completely, including this variable and the constructor.

But I'd like to confirm with @mavam before doing that since he's the original author. So maybe we can merge the "safe" fix now and leave a TODO for the refactoring?

Reply:
I think in the scope of this fix, we can remove the member entirely.

Reply:
Regarding removing `native_type`, let's wait with that because I don't think it's unreasonable to request the file descriptor from a file abstraction, e.g., to call `flock(2)` or other things on it.